lib/internal/source_map/source_map_cache.js

```diff
@@ -205,14 +203,26 @@ function dataFromUrl(sourceURL, sourceMappingURL) {
 // from. This allows translation from byte offset V8 coverage reports,
 // to line/column offset Source Map V3.
 function lineLengths(content) {
-  // We purposefully keep \r as part of the line-length calculation, in
-  // cases where there is a \r\n separator, so that this can be taken into
-  // account in coverage calculations.
-  return ArrayPrototypeMap(RegExpPrototypeSymbolSplit(/\n|\u2028|\u2029/, content), (line) => {
-    return line.length;
-  });
+  const contentLength = content.length;
+  const output = [];
+  let lineLength = 0;
+  for (let i = 0; i < contentLength; i++, lineLength++) {
+    const codePoint = content.codePointAt(i);
+
+    // We purposefully keep \r as part of the line-length calculation, in
+    // cases where there is a \r\n separator, so that this can be taken into
+    // account in coverage calculations.
+    // codepoints for \n, \u2028 and \u2029
+    if (codePoint === 10 || codePoint === 0x2028 || codePoint === 0x2029) {
+      output.push(lineLength);
+      lineLength = -1; // To not count the matched codePoint such as \n character
```

Comment:
Let’s say you forgot this line (`lineLength = -1`). Is there a test that would fail? If not, we should add one.

We can’t really add a test for “ensure that this doesn’t use as much memory as before” but if we don’t have adequate coverage that this logic is behaving as expected, this is a good opportunity to improve our coverage.

Reply:
All the existing tests that test this codepath will fail as the line lengths end up being calculated incorrectly. I tested the failure locally as well!

Reply:
For example — 
<img width="982" alt="image" src="https://github.com/nodejs/node/assets/635512/6e5c5c53-b92e-4ed9-a64c-2be3f87a9119">